unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/VisibilityComponent/Handler/ECSVisibilityComponentHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using DCL.Controllers;
+using DCL.ECS7.InternalComponents;
+using DCL.ECSRuntime;
+using DCL.Models;
+using UnityEngine;
+
+namespace DCL.ECSComponents
+{
+    public class ECSVisibilityComponentHandler : IECSComponentHandler<PBVisibilityComponent>
+    {
+        public ECSVisibilityComponentHandler() { }
+
+        private void SetVisibility(IDCLEntity entity, bool visible)
+        {
+            MeshesInfo meshesInfo = entity.meshesInfo;
+            
+            ShapeRepresentation shape = meshesInfo.currentShape as ShapeRepresentation;
+            shape?.UpdateModel(visible, shape.HasCollisions());
+            
+            ECSComponentsUtils.ConfigurePrimitiveShapeVisibility(entity.gameObject, visible, meshesInfo.renderers);
```

Comment:
There is a potential issue with this approach:
if a"visibility" component is put before some "rendereable" component, the visibility won't work.
I would suggest doing something like we are currently doing with materials.
Meaning every component with renderers could modify an internal component holding the renderers, and a system checking a group of entities with renderers and visibility components, so if any of those component changed we could apply those changes there